library/src/scala/TupleXXL.scala

```diff
@@ -1,35 +1,35 @@
 package scala
 import java.util.Arrays.{deepEquals, deepHashCode}
 
-final class TupleXXL private (es: Array[Object]) extends Product {
+final class TupleXXL private (es: IArray[Object]) extends Product {
   assert(es.length > 22)
 
   def productElement(n: Int): Any = es(n)
   def productArity: Int = es.length
 
-  override def toString = elems.mkString("(", ",", ")")
-  override def hashCode = getClass.hashCode * 41 + deepHashCode(elems)
+  override def toString = elems.asInstanceOf[Array[Object]].mkString("(", ",", ")")
+  override def hashCode = getClass.hashCode * 41 + deepHashCode(elems.asInstanceOf[Array[Object]])
   override def canEqual(that: Any): Boolean = that match {
     case that: TupleXXL => that.productArity == this.productArity
     case _ => false
   }
 
   override def equals(that: Any) = that match {
-    case that: TupleXXL => deepEquals(this.elems, that.elems)
+    case that: TupleXXL => deepEquals(this.elems.asInstanceOf[Array[Object]], that.elems.asInstanceOf[Array[Object]])
     case _ => false
   }
-  def elems: Array[Object] = es
+  def elems: IArray[Object] = es
 
   def tailXXL: TupleXXL = {
     assert(es.length > 23)
-    new TupleXXL(es.tail)
+    new TupleXXL(es.asInstanceOf[Array[Object]].tail.asInstanceOf[IArray[Object]])
   }
 
-  def toArray: Array[Object] = es.clone
+  def toArray: Array[Object] = es.asInstanceOf[Array[Object]].clone
 }
 object TupleXXL {
-  def fromIterator(elems: Iterator[Any]) = new TupleXXL(elems.map(_.asInstanceOf[Object]).toArray)
-  def apply(elems: Array[Object]) = new TupleXXL(elems.clone)
-  def apply(elems: Any*) = new TupleXXL(elems.asInstanceOf[Seq[Object]].toArray)
-  def unapplySeq(x: TupleXXL): Option[Seq[Any]] = Some(x.elems.toSeq)
+  def fromIterator(elems: Iterator[Any]) = new TupleXXL(elems.map(_.asInstanceOf[Object]).toArray.asInstanceOf[IArray[Object]])
```

Comment:
It's better to avoid `.asInstanceOf[IArray[Object]]`, as the Array instance might be aliased, thus break the protocol. Create `IArray` through its constructors is recommended.

Reply:
`IArray` would copy the array at least twice, this is the best we can do for now. We need `Iterator.toIArray`.